### PR TITLE
Np 664 update java and sar

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: FindChannelFunction/build.gradle [hunk 1]
```diff
@@ -5,15 +5,19 @@ plugins {
...
     jcenter()
 }
 
 
+
 dependencies {
     implementation 'com.amazonaws:aws-lambda-java-core:1.2.0'
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.1'
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.10.1'
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
```
Comment (axthosarouris) [inline FindChannelFunction/build.gradle:17]: create a variable and use the variable 
```suggestion
def jacksonVersion = "2.10.3"

dependencies{
implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
...
}
        implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
```

**Review by axthosarouris — DISMISSED** [review]
Just a small change

File: FindChannelFunction/build.gradle [hunk 2]
```diff
@@ -37,8 +41,10 @@ pmd {
 checkstyle {
     configFile = rootProject.resources.text.fromFile('config/checkstyle/checkstyle.xml').asFile()
     showViolations = true
+    toolVersion "8.30"
```
Comment (janbj) [inline FindChannelFunction/build.gradle:42]: Why do we set version?

File: buildspec.yaml [hunk 3]
```diff
@@ -4,14 +4,14 @@ phases:
...
-      java: openjdk8
+      java: openjdk11
 
     commands:
       - pip3 install aws-sam-cli
 
   build:
     commands:
-      - sam build
+      - sam build --debug
```
Comment (janbj) [inline buildspec.yaml]: Do we always want debug here?

File: template.yaml [hunk 4]
```diff
@@ -40,6 +40,7 @@ Resources:
     Type: AWS::Serverless::Api
     Properties:
       StageName: Prod
+      EndpointConfiguration: REGIONAL
```
Comment (janbj) [inline template.yaml:42]: What's the advantage of REGIONAL again?

**Review by janbj — DISMISSED** [review]
Approved, just a few questions.

File: FindChannelFunction/build.gradle (hunk 1 above)
Comment (brinxmat) [reply FindChannelFunction/build.gradle:17]: But use group: ... formatting please.

**Review by brinxmat — DISMISSED** [review]

File: template.yaml (hunk 4 above)
Comment (svenngjeto) [reply template.yaml:42]: An edge-optimized API endpoint is best for geographically distributed clients. API requests are routed to the nearest CloudFront Point of Presence (POP). This is the default endpoint type for API Gateway REST APIs.
A regional API endpoint is intended for clients in the same region. When a client running on an EC2 instance calls an API in the same region, or when an API is intended to serve a small number of clients with high demands, a regional API reduces connection overhead. .

File: FindChannelFunction/build.gradle (hunk 2 above)
Comment (svenngjeto) [reply FindChannelFunction/build.gradle:42]: To make i compile... Checkstyle was not happy with default...

File: buildspec.yaml (hunk 3 above)
Comment (svenngjeto) [reply buildspec.yaml]: Unnecessary, removed!

File: FindChannelFunction/build.gradle (hunk 1 above)
Comment (svenngjeto) [reply FindChannelFunction/build.gradle:17]: Reformatted

**Pull request merged**